tests/test_multipart.py

```diff
@@ -67,6 +68,8 @@ def __init__(self, content):
 
     @asyncio.coroutine
     def read(self, size=None):
+        if size is not None:
+            size = random.randint(size // 2, size)
```

Comment:
I dislike the test.
Let's assume we will break the code again.
Looking on random test failures I'll have no clue what's broken.